lib/Transforms/ConvertIfToSelect/ConvertIfToSelect.td

```diff
@@ -0,0 +1,14 @@
+#ifndef LIB_TRANSFORMS_CONVERTIFTOSELECT_CONVERTIFTOSELECT_TD_
+#define LIB_TRANSFORMS_CONVERTIFTOSELECT_CONVERTIFTOSELECT_TD_
+
+include "mlir/Pass/PassBase.td"
+
+def ConvertIfToSelect : Pass<"convert-if-to-select"> {
+  let summary = "Convert If-operations on secret conditions to select operations.";
+  let description = [{ Conversion for If-operations that evaluate secret condition(s) to their alternative select operations. }];
+  let dependentDialects = [
+    "mlir::scf::SCFDialect",
```

Comment:
dependentDialects needs to include the dialects of (new) operations the pass might introduce (see https://mlir.llvm.org/docs/PassManagement/#dependent-dialects).
Without this, a program that has an `scf.if` but not yet any `arith` ops could crash when the pass tries to create the `arith.select` op, due to Arith not being loaded yet.
Technically, I think that means SCF isn't even required (if there ar no `scf` ops in the program, the pass doesn't do anything) but it probably doesn't hurt to force it being loaded.

```suggestion
    "mlir::scf::SCFDialect",
    "mlir::arith::ArithDialect"
```